lib/bunny/session.rb

```diff
@@ -191,7 +191,7 @@ def initialize(connection_string_or_opts = ENV['RABBITMQ_URL'], optz = Hash.new)
 
       client_props         = opts[:properties] || opts[:client_properties] || {}
       @client_properties   = DEFAULT_CLIENT_PROPERTIES.merge(client_props)
-      @mechanism           = normalize_auth_mechanism(opts.fetch(:auth_mechanism, "PLAIN"))
+      @mechanism           = (opts[:auth_mechanism] || []).first || "PLAIN"
```

Comment:
If we modify this at all (not sure why it's a part of this PR but OK), `normalize_auth_mechanism` should then be deleted.

Reply:
Yeah, nice spot, I forgot to remove the method. Basically `auth_mechanism` confused me too. According to specification client may specify several auth mechanism, so I decided to adheer in `amq-protocol`. But when I proceed with `bunny`, I realised that session initialization expects a single string value, not array. TBH, I didn't have time to analyze, how several auth mechanisms should be handled properly, so i decided to adheer bunny fallback to `"PLAIN"` for array case.

Let me recheck, maybe `normalize_auth_mechanism` is OK for array default value, I don't remember exact reason, why I replaced it...